src/Bicep.Core.IntegrationTests/ScenarioTests.cs

```diff
@@ -3115,6 +3115,38 @@ public void Test_Issue_3169_3()
 }
 
 resource existingStg 'Microsoft.Storage/storageAccounts@2021-04-01' existing = {
+  name: newStg.name
+}
+
+resource newStg2 'Microsoft.Storage/storageAccounts@2021-04-01' = {
+  name: existingStg.name
+  kind: 'StorageV2'
+  location: resourceGroup().location
+  sku: {
+    name: 'Standard_LRS'
+  }
+}
+");
+            result.ExcludingLinterDiagnostics().Should().NotHaveAnyDiagnostics();
```

Comment:
Can we add an assertion to verify if `newStd2.name` is `test` in the compiled JSON template?

Reply:
So turns out the json template does not carry over the names of the resource such as `newStg2` so what I did was give `newStg2` a different storage type "BlobStorage" than that of `newStg` which is `StorageV2`. From there I added an assertion that the resource with storage type "BlobStorage" had the DTC value we are looking for.